From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Spottedleaf <spottedleaf@spottedleaf.dev>
Date: Sun, 10 May 2020 06:23:56 -0700
Subject: [PATCH] Optimise sync loads for getType calls

We don't need a full chunk, just EMPTY. This will prevent us
from having to load its neighbours.

diff --git a/src/main/java/net/minecraft/server/World.java b/src/main/java/net/minecraft/server/World.java
index c904e22281cb99c90e09e5a4fd3241b450f17ab8..cf6b435dd19a0fa12dbb7ceb94bad0685332ff93 100644
--- a/src/main/java/net/minecraft/server/World.java
+++ b/src/main/java/net/minecraft/server/World.java
@@ -662,7 +662,14 @@ public abstract class World implements GeneratorAccess, AutoCloseable {
         if (isOutsideWorld(blockposition)) {
             return Blocks.VOID_AIR.getBlockData();
         } else {
-            Chunk chunk = this.getChunkAt(blockposition.getX() >> 4, blockposition.getZ() >> 4);
+            // Tuinity start
+            int chunkX = blockposition.getX() >> 4;
+            int chunkZ = blockposition.getZ() >> 4;
+            IChunkAccess chunk = this.getChunkIfLoadedImmediately(chunkX, chunkZ);
+            if (chunk == null) {
+                chunk = this.getChunkAt(chunkX, chunkZ, ChunkStatus.EMPTY, true);
+            }
+            // Tuinity end
 
             return chunk.getType(blockposition);
         }
